include/stl2/detail/algorithm/sort.hpp

```diff
@@ -24,81 +24,125 @@
 // sort [sort]
 //
 STL2_OPEN_NAMESPACE {
-	template<RandomAccessIterator I, Sentinel<I> S, class Comp = less,
-		class Proj = identity>
-	requires
-		Sortable<I, Comp, Proj>
-	I sort(I first, S sent, Comp comp = {}, Proj proj = {})
-	{
-		if (first == sent) {
-			return first;
+	struct __sort_fn : private __niebloid {
+	private:
+		static constexpr std::ptrdiff_t introsort_threshold = 16;
+
+		template<RandomAccessIterator I, class Comp, class Proj>
+		requires Sortable<I, Comp, Proj>
+		static constexpr I unguarded_partition(I first, I last, Comp& comp, Proj& proj)
+		{
+			I pivot_pnt = detail::rsort::choose_pivot(first, last, comp, proj);
+
+			// Do the partition:
+			while (true) {
+				auto &&v = *pivot_pnt;
+				auto &&pivot = __stl2::invoke(proj, (decltype(v) &&)v);
+				while (__stl2::invoke(comp, __stl2::invoke(proj, *first), pivot)) {
+					++first;
+				}
+				while (__stl2::invoke(comp, pivot, __stl2::invoke(proj, *--last))) {
+					[[fallthrough]]; // there's no [[do_nothing]] attribute
```

Comment:
http://eel.is/c++draft/dcl.attr.fallthrough#1 "The attribute-token fallthrough may be applied to a null statement; such a statement is a fallthrough statement. ... A fallthrough statement may only appear within an enclosing `switch` statement."
```suggestion
					; // do nothing
```